src/agent/src/main.rs

```diff
@@ -354,6 +378,72 @@ async fn start_sandbox(
     Ok(())
 }
 
+// If we fail to start the AA, ocicrypt won't be able to unwrap keys
+// and container decryption will fail.
+fn init_attestation_agent(logger: &Logger) -> Result<()> {
+    let config_path = OCICRYPT_CONFIG_PATH;
+
+    // The image will need to be encrypted using a keyprovider
+    // that has the same name (at least according to the config).
+    let ocicrypt_config = serde_json::json!({
+        "key-providers": {
+            "attestation-agent":{
+                "ttrpc":AA_KEYPROVIDER_URI
+            }
+        }
+    });
+
+    fs::write(config_path, ocicrypt_config.to_string().as_bytes())?;
+
+    env::set_var("OCICRYPT_KEYPROVIDER_CONFIG", config_path);
+
+    let attestation_sock = Path::new(AA_ATTESTATION_SOCKET);
+    if attestation_sock.exists() {
+        fs::remove_file(AA_ATTESTATION_SOCKET)?;
+    }
+
+    // The Attestation Agent will run for the duration of the guest.
+    Command::new(AA_PATH)
+        .arg("--keyprovider_sock")
+        .arg(AA_KEYPROVIDER_URI)
+        .arg("--getresource_sock")
+        .arg(AA_GETRESOURCE_URI)
+        .arg("--attestation_sock")
+        .arg(AA_ATTESTATION_URI)
+        .spawn()?;
+
+    // wait attestation-agent boot
+    for _ in 0..20 {
+        if attestation_sock.exists() {
+            break;
+        }
+        info!(logger, "waiting for attestation-agent boot ...");
+        std::thread::sleep(std::time::Duration::from_secs(1));
+    }
+
+    #[cfg(feature = "confidential-data-hub")]
+    {
+        // launch CDH
+        let cdh_socket = Path::new(CDH_SOCKET);
+        if cdh_socket.exists() {
+            fs::remove_file(CDH_SOCKET)?;
+        }
+
+        Command::new(CDH_PATH).spawn()?;
+
+        // wait CDH boot
+        for _ in 0..20 {
+            if cdh_socket.exists() {
+                break;
+            }
+            info!(logger, "waiting for confidential-data-hub boot ...");
+            std::thread::sleep(std::time::Duration::from_secs(1));
+        }
```

Comment:
Should we add some kind of catch here to log an error if we never find the socket?

Reply:
good suggestion,  I will update.